src/attr/_make.py

```diff
@@ -709,13 +709,39 @@ def _make_hash(attrs):
         if a.hash is True or (a.hash is None and a.cmp is True)
     )
 
-    def hash_(self):
-        """
-        Automatically created by attrs.
-        """
-        return hash(_attrs_to_tuple(self, attrs))
+    # We cache the generated init methods for the same kinds of attributes.
+    sha1 = hashlib.sha1()
+    sha1.update(repr(attrs).encode("utf-8"))
+    unique_filename = "<attrs generated hash {0}>".format(
+        sha1.hexdigest()
+    )
+
+    lines = [
+        "def __hash__(self):",
+        "    return hash((",
+        "        self.__class__,",
```

Comment:
Yeah we should totally cache the value…as for random, it's kinda hard to use the full spectrum since we’ve lost maxint and I’m not aware of any replacements…so maybe just `random.random() * random.choice((1.0, -1.0))`?